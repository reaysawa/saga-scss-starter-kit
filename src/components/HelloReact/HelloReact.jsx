import React from 'react';

import styles from './styles';

function HelloReact(props) {
  return (
    <div className={`${styles['title']}`}>Hello {props.name}</div>
  );
}

export default HelloReact
