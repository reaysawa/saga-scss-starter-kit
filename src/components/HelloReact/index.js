import React, { useState } from 'react';

import * as styles from './styles';

function HelloReact() {
  const [message, setMessage] = useState('Nathan')
  return (
    <div className={styles['title']}>Hello {message}</div>
  );
}

export default HelloReact
