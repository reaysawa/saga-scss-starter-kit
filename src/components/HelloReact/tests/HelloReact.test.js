import React from "react"
import { shallow } from "enzyme"

import HelloReact from "../HelloReact"

describe("HelloReact renders", () => {
  it("the name Charles", () => {
    const wrapper = shallow(<HelloReact name={"Charles"} />)

    const charlesWasRendered = wrapper.html().indexOf("Charles") > -1
    expect(charlesWasRendered).toBeTruthy()
    expect(wrapper).toMatchSnapshot()
  })
})
