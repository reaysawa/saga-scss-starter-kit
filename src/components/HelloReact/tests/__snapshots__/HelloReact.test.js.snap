// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloReact renders the name Charles 1`] = `
<div
  className="title"
>
  Hello 
  Charles
</div>
`;

exports[`HelloReact renders with the name prop renders the name Charles 1`] = `
<div
  className="title"
>
  Hello 
  Charles
</div>
`;
